### DISCUSSION TITLE
How to restore back to older working versions?

### DISCUSSION THREAD

**Discussion by ebguy06** [aclap-dev/video-downloadhelper#242] [Help and Q&A]
### Your question

How to restore back to older working versions?

I assume I do the following, what am I missing?

Turn off automatic updates on downloadhelper extension.

Download the older WORKING version.
`https://addons.mozilla.org/en-US/firefox/addon/video-downloadhelper/versions/`
Version 8.1.4.0
Released Jan 4, 2024 - 884.54 KB

What  VdhCoApp should i use?
Do I need to use an older Firefox version from that time period?

Anything else?

Was this all a ruse, a conspiracy to get people to purchase a [premium] license in order to HLS stream downloads which were working great (Version 8.1.4.0)? because it feels like even the older versions have been tampered with and no longer work either.
Sorry so exhausted, two full days of trying to get this to work and cant record my paid services.

### Addon & CoApp version and details

`version: 8.2.2.0a1
channel: beta
build date: 2024-03-01
build options: linuxlic: false, noyt: false, browser: firefox
lang: en-US
license: unset
platform: x86-64 win
UA: Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:123.0) Gecko/20100101 Firefox/123.0
coapp: yes. Info:
{
  "id": "net.downloadhelper.coapp",
  "name": "VdhCoApp",
  "version": "2.0.19",
  "binary": "C:\\Program Files\\DownloadHelper CoApp\\vdhcoapp.exe",
  "displayName": "VdhCoApp",
  "description": "Video DownloadHelper companion app",
  "target": {
    "os": "windows",
    "arch": "x86_64",
    "node": "18"
  },
  "home": "C:\\Users\\oneof",
  "converterBinary": "C:\\Program Files\\DownloadHelper CoApp\\ffmpeg.exe",
  "converterBase": "ffmpeg",
  "converterBaseVersion": "c30f360955"
}
`

**Comment by Mattmjs** [comment]
I'm just another user, for 8.14.0 it says companion app version 2.0.10 is required. No change needed on Firefox version.
You don't necessarily need a license, however it will add a watermark on places like YouTube. 
From 2.0.10 to 2.0.19 , you might find the right combination to go with 8.1.4.0

**Comment by paulrouget** [comment]
Normally all coapps are compatible with 8.1.

What's going on with 8.2?

**Reply by Mattmjs** [reply]
https://github.com/aclap-dev/video-downloadhelper/discussions/236#discussioncomment-8649304

https://github.com/aclap-dev/video-downloadhelper/discussions/240#discussioncomment-8649138

**Comment by spookycat567** [comment]
do you have the older working version for Chrome? mine isn't working

**Reply by MarkoIndaco** [reply]
Still looking for old version for Chrome. Anyone?

**Reply by GrampaWildWilly** [reply]
I do not work for VDH.  I am just another user.

There is no archive of older versions of Chrome (nor Edge) the way there is for Firefox.  Mozilla themselves keep that archive.  But Google (nor Microsoft) does not maintain such an archive, nor is there such a repository on the VDH web site nor here on GitHub.  You are forced to do web searches for this.  I vaguely recall looking for an older version of something a few years ago & web searches turned up several archives of old versions of many applications.  You're going to have to do some scrounging.  Sad but true.

**Reply by Mattmjs** [reply]
It's easier to install older versions in Firefox and disable auto updates, but if you insist you need an old version for chrome
well someone found a site. Refer to this #1245 

You need to understand that there is no switch to disable extensions from auto updating in chrome. I left a link to stackoverflow
in that discussion.

**Reply by bushnest** [reply]
There is a way to downgrade to older versions of VDH on Chrome.  First, uninstall the extension.  Next, download the crx version of the version of VDH that you want installed in Chrome.  Next open up a terminal in the folder you downloaded the crx file into.  Then, in the terminal, type <code>unzip *.crx</code>.  You should now have the crx file unzipped into its very own folder.  Now, go to that folder in your terminal.  Then, type <code>nano manifest.json</code>.  After opening manifest.json in nano, do a search for the phrase 'update_url'.  After doing that, the phrase will appear along with a URL right behind that.  Now, in your text editor, change that URL to whatever BS URL that you can think of.  Save the changes you made to manifest.json.  Next, you will need to zip that entire folder up.  To do that type <code>zip -r (zip file name) (directory name)</code>.  Now, open up Chrome and open the menu by clicking in the upper right-hand corner.  Select 'Extensions'.  Then, click on  '+(from .zip/.crx/user js).  Search for the .zip file you created and click on it.  VDH of the version you want will be installed and will not be able to autoupdate.  Of course, it will spit out error messages every time it tries to update, but you can just ignore those error messages   It's basically just the extension crying about it not being able to update.  There you have it - VDH installed in the version you want and unable to autoupdate.
